lib/board.rb

```diff
@@ -10,4 +10,12 @@ def place_token(pos, token)
   def get(pos)
     @board[pos - 1]
   end
+
+  def next_token
+    @board.count("X") > @board.count("O") ? "O" : "X"
```

Comment:
It might be nice to start extracting strings into constants since we're using them in a couple of different places in the next iteration. I even like to put messages, like the prompt, "Please select your move by entering the number (1 - 9, from top left to bottom right) corresponding to an empty space."  into constants with good, clear names. Strings are prone to typos, and if there is ever an occasion where you want to change your symbol to something else, it's nice to be able to go to one place and change it instead of finding all the different places where you used the string to replace them individually.

Reply:
Good point.  Thanks

Reply:
Additionally, this might be of use to you
https://ruby-doc.org/core-2.7.0/Array.html#method-i-cycle

Reply:
I set up a tokens hash as an instance variable in the board class to manage tokens

Reply:
I did the same for player messages